saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/SaulWekaWrapper.java

```diff
@@ -0,0 +1,587 @@
+    }
+
+
+    /**
+     * Produces a set of scores indicating the degree to which each possible discrete classification
+     * value is associated with the given example object.
+     **/
+    public ScoreSet scores(int[] exampleFeatures, double[] exampleValues) {
+        double[] dist = getDistribution(exampleFeatures, exampleValues);
+
+        /*
+         * Assuming that the first Attribute in our attributeInfo vector is the class attribute,
+         * decide which case we are in
+         */
+        Attribute classAtt = (Attribute) attributeInfo.elementAt(0);
+
+        ScoreSet scores = new ScoreSet();
+
+        if (classAtt.isNominal() || classAtt.isString()) {
+            Enumeration enumeratedValues = classAtt.enumerateValues();
+
+            int i = 0;
+            while (enumeratedValues.hasMoreElements()) {
+                if (i >= dist.length) {
+                    System.err
+                            .println("WekaWrapper: Error - scores found more possible values than "
+                                    + "probabilities.");
+                    new Exception().printStackTrace();
+                    System.exit(1);
+                }
+                double s = dist[i];
+                String v = (String) enumeratedValues.nextElement();
+                scores.put(v, s);
+                ++i;
+            }
+        } else if (classAtt.isNumeric()) {
+            System.err.println("WekaWrapper: Error - The 'scores' function should not be called "
+                    + "when the class attribute is numeric.");
+            new Exception().printStackTrace();
+            System.exit(1);
+        } else {
+            System.err.println("WekaWrapper: Error - ScoreSet: Class Types must be either "
+                    + "Nominal, String, or Numeric.");
+            new Exception().printStackTrace();
+            System.exit(1);
+        }
+
+        return scores;
+    }
+
+
+    /**
+     * Indicates that the classifier is finished learning. This method <I>must</I> be called if the
+     * WEKA classifier is to learn anything. Since WEKA classifiers cannot learn online, all of the
+     * training lbjavaInstances must be gathered and committed to first. This method invokes the WEKA
+     * classifier's <code>buildClassifier(Instances)</code> method.
+     **/
+    public void doneLearning() {
+        if (trained) {
+            System.err.println("WekaWrapper: Error - Cannot call 'doneLearning()' again without "
+                    + "first calling 'forget()'");
+            new Exception().printStackTrace();
+            System.exit(1);
+        }
+
+        /*
+         * System.out.println("\nWekaWrapper Data Summary:");
+         * System.out.println(wekaInstances.toSummaryString());
+         */
+
+        try {
+            initializeAttributes();
+            for (Tuple4<int[], double[], int[], double[]> t : lbjavaInstances)
+                wekaInstances.add(makeInstance(t.t1(), t.t2(), t.t3(), t.t4()));
```

Comment:
Should we clear the items in `lbjavaInstances` after adding them to `wekaInstances`? We will have two copies of data in memory otherwise. Thoughts?